certbot/client.py

```diff
@@ -317,23 +318,30 @@ def save_certificate(self, certr, chain_cert,
 
         cert_pem = OpenSSL.crypto.dump_certificate(
             OpenSSL.crypto.FILETYPE_PEM, certr.body.wrapped)
-        cert_file, act_cert_path = le_util.unique_file(cert_path, 0o644)
+
+        cert_file, abs_cert_path = _open_pem_file('cert_path', cert_path)
+
         try:
             cert_file.write(cert_pem)
         finally:
             cert_file.close()
         logger.info("Server issued certificate; certificate written to %s",
-                    act_cert_path)
+                    abs_cert_path)
 
-        cert_chain_abspath = None
-        fullchain_abspath = None
-        if chain_cert:
+        if not chain_cert:
+            return abs_cert_path, None, None
```

Comment:
I'd personally like to see `certbot` at 100% coverage too. I feel it is kind of necessary with Python to prevent problems like #2822 and #3012, but I'm not worried about it here. If you feel like writing a follow up PR adding test coverage though, I'll review and merge it.